plonky3/src/circuit_builder.rs

```diff
@@ -48,19 +57,36 @@ pub fn cast_to_goldilocks<T: FieldElement>(v: T) -> Val {
     Val::from_canonical_u64(v.to_integer().try_into_u64().unwrap())
 }
 
+fn get_publics<T: FieldElement>(analyzed: &Analyzed<T>) -> Vec<(String, usize)> {
+    let mut publics = analyzed
+        .public_declarations
+        .values()
+        .map(|public_declaration| {
+            let witness_name = public_declaration.referenced_poly_name();
+            let witness_offset = public_declaration.index as usize;
+            (witness_name, witness_offset)
+        })
+        .collect::<Vec<_>>();
+
+    // Sort, so that the order is deterministic
+    publics.sort();
+    publics
+}
+
 impl<'a, T: FieldElement> PowdrCircuit<'a, T> {
     pub(crate) fn new(analyzed: &'a Analyzed<T>) -> Self {
         if analyzed.constant_count() > 0 {
             unimplemented!("Fixed columns are not supported in Plonky3");
         }
 
-        if !analyzed.public_declarations.is_empty() {
-            unimplemented!("Public declarations are not supported in Plonky3");
-        }
+        // if !analyzed.public_declarations.is_empty() {
+        //     unimplemented!("Public declarations are not supported in Plonky3");
+        // }
 
         Self {
             analyzed,
             witness: None,
+            publics: get_publics(analyzed),
```

Comment:
Seems like duplicated data, instead we could turn `get_publics` into a method on `PowdrCircuit` which computes this on the fly